### PR TITLE
adding context variables to onEvent session aggregate

### PR DESCRIPTION
**Pull request by vder**

### CODE REVIEW

File: engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/OnEventTriggerWindowOperator.scala [hunk 1]
```diff
@@ -24,7 +24,7 @@ import pl.touk.nussknacker.engine.flink.util.transformer.aggregate.triggers.Fire
 import java.lang
 
 object OnEventTriggerWindowOperator {
-  type Input = ValueWithContext[StringKeyedValue[AnyRef]]
+  type Input[A] = ValueWithContext[StringKeyedValue[A]]
```
Comment (lciolecki) [inline engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/OnEventTriggerWindowOperator.scala:27]: A -> T :D It's kind of standard, I think.. ;p

**Review by lciolecki — APPROVED** [review]
LGTM, Let @gadomsky  have a look at it

File: engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/transformers.scala [hunk 2]
```diff
@@ -112,28 +112,32 @@ object transformers {
                                sessionWindowTrigger: SessionWindowTrigger,
                                variableName: String
                               )(implicit nodeId: NodeId): ContextTransformation =
-    // TODO: to be consistent with sliding window we should probably forward context of variables for tumblingWindowTrigger == SessionWindowTrigger.OnEnd
-    ContextTransformation.definedBy(aggregator.toContextTransformation(variableName, emitContext = false, aggregateBy))
+  // TODO: to be consistent with sliding window we should probably forward context of variables for tumblingWindowTrigger == SessionWindowTrigger.OnEnd
```
Comment (jedrz) [inline engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/transformers.scala:115]: I think we can remove the comment safely :wink:

**Review by jedrz — APPROVED** [review]
:+1:

File: engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/transformers.scala (hunk 2 above)
Comment (vder) [reply engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/transformers.scala]: removed

File: docs/scenarios_authoring/AggregatesInTimeWindows.md [hunk 3]
```diff
@@ -19,9 +19,9 @@ Nussknacker implements 3 types of time windows - tumbling, sliding and session w
...
+Other window types - tumbling window in `After window closes` emit mode, and session window in `After session end` nodes behave differently - they generate a new event containing an aggregate when the time window is closed. Its timestamp is equal to the time of the timer that generated it, not system time of the moment when it happened. In other words the timestamp of the newly generated event which contains the aggregate will continue to use the notion of time used by events which the aggregate window saw. These nodes 'terminate' the events which entered them; consequently all the variables defined before the node (including `#input` and `#inputMeta`) will not be available downstream. The sliding-window behaves in the same way in its **non-default** configuration when parameter `emitWhenEventLeft` is set to `true` - it terminates events entering the aggregation node, emits the aggregate event when window is closed.
```
Comment (ZbyszekMM) [inline docs/scenarios_authoring/AggregatesInTimeWindows.md:24]: in `After session end` nodes -> in `After session end` mode

File: docs/scenarios_authoring/AggregatesInTimeWindows.md [hunk 4]
```diff
@@ -19,9 +19,9 @@ Nussknacker implements 3 types of time windows - tumbling, sliding and session w
...
 
-A sliding-window in its default configuration (`emitWhenEventLeft` is set to `false`) and a tumbling-window with `On each event` emit mode enrich the incoming event with the value of the aggregate. If there were no aggregates "upstream" which terminated the events which entered scenario, the  events which entered the scenario will still be available downstream. As the variables are "attached" to the events, the `#input` and `#inputMeta` variables will be also available 'downstream'. 
+A sliding-window in its default configuration (`emitWhenEventLeft` is set to `false`) and a window with `On each event` emit mode (either tumbling-window or session-window) enrich the incoming event with the value of the aggregate. If there were no aggregates "upstream" which terminated the events which entered scenario, the  events which entered the scenario will still be available downstream. As the variables are "attached" to the events, the `#input` and `#inputMeta` variables will be also available 'downstream'.
```
Comment (ZbyszekMM) [inline docs/scenarios_authoring/AggregatesInTimeWindows.md:22]: May be a bit easier to follow:

and a window with `On each event` emit mode (either tumbling-window or session-window) ->
and both tumbling-window and session-window in `On each event` emit mode

File: docs/scenarios_authoring/AggregatesInTimeWindows.md (hunk 3 above)
Comment (vder) [reply docs/scenarios_authoring/AggregatesInTimeWindows.md]: changed

File: docs/scenarios_authoring/AggregatesInTimeWindows.md (hunk 4 above)
Comment (vder) [reply docs/scenarios_authoring/AggregatesInTimeWindows.md]: changed

**Pull request merged**